bundles/persistence/org.openhab.persistence.sitewhere/src/main/java/org/openhab/persistence/sitewhere/internal/SiteWherePersistenceService.java

```diff
@@ -0,0 +1,443 @@
+
+	/** Alert type indicator for a DateTimeType */
+	public static final String TYPE_OPENHAB_DATETIME = "openhab.datetime";
+
+	/** Alert type indicator for a OnOffType */
+	public static final String TYPE_OPENHAB_ONOFF = "openhab.onoff";
+
+	/** Alert type indicator for a OpenClosedType */
+	public static final String TYPE_OPENHAB_OPENCLOSED = "openhab.openclosed";
+
+	/** Alert type indicator for a StringType */
+	public static final String TYPE_OPENHAB_STRING = "openhab.string";
+
+	/** Service name */
+	private static final String SERVICE_NAME = "sitewhere";
+
+	/** REST client used to interact with SiteWhere */
+	private ISiteWhereClient client;
+
+	/** SiteWhere MQTT agent */
+	private Agent agent;
+
+	/** Command processor that interacts with SiteWhere */
+	private OpenHabCommandProcessor sitewhere;
+
+	/** Reference to openHAB item registry */
+	protected ItemRegistry itemRegistry;
+
+	/** Event publisher for openHAB */
+	private EventPublisher eventPublisher;
+
+	/** Indicates if the service has been configured */
+	protected boolean configured = false;
+
+	/** Indicates if REST connectivity has been verified */
+	protected boolean restVerified = false;
+
+	/** Indicates if SiteWhere agent is connected */
+	protected boolean agentConnected = false;
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.openhab.core.persistence.PersistenceService#getName()
+	 */
+	public String getName() {
+		return SERVICE_NAME;
+	}
+
+	/**
+	 * OSGI setter for {@link EventPublisher}.
+	 * 
+	 * @param publisher
+	 */
+	public void setEventPublisher(EventPublisher publisher) {
+		this.eventPublisher = publisher;
+	}
+
+	/**
+	 * OSGI unsetter for {@link EventPublisher}.
+	 * 
+	 * @param publisher
+	 */
+	public void unsetEventPublisher(EventPublisher publisher) {
+		this.eventPublisher = null;
+	}
+
+	/**
+	 * Method called when bundle is activated.
+	 */
+	public void activate() {
+		LOGGER.debug("Activate method called on persistence service.");
+		try {
+			SiteWhereConfigurationManager.getInstance();
+			setConfigured(true);
+			validateSiteWhereClient();
+			validateSiteWhereAgent();
+		} catch (Throwable t) {
+			LOGGER.error("SiteWhere activation failed.", t);
+			setConfigured(false);
+		}
+	}
+
+	/**
+	 * Method called when bundle is deactivated.
+	 */
+	public void deactivate() {
+		LOGGER.debug("Deactivate method called on persistence service.");
+		setConfigured(false);
+	}
+
+	/**
+	 * Set reference to item registry.
+	 * 
+	 * @param itemRegistry
+	 */
+	public void setItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = itemRegistry;
+		LOGGER.debug("Item registry set for SiteWhere persistence.");
+	}
+
+	/**
+	 * Unset reference to item registry.
+	 * 
+	 * @param itemRegistry
+	 */
+	public void unsetItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = null;
+	}
+
+	/**
+	 * Validate that client is able to interact with SiteWhere server.
+	 * 
+	 * @return
+	 */
+	protected boolean validateSiteWhereClient() {
```

Comment:
since this is the method where the client is also created i would rename this to ```connect()``` or alike and explain in javadoc that the clients' connection is also tested/verified